<?xml version="1.0"?>
<launch>
  <arg name="num_dataset" default="100" />

 <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find gen_dataset)/worlds/empty.world"/>
  </include>
  <node name="setup_commands" pkg="timed_roslaunch" type="timed_roslaunch.sh" args="3 gen_dataset setup_commands.launch" output="screen" />

  <arg name="gui" default="true"/>
  <param name="use_gui" value="$(arg gui)"/>
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find gen_dataset)/rviz/generator_config.rviz" required="true" />
  <include file="$(find gen_dataset)/launch/gen_multi_photoneo.launch" />

  <group ns="p1">
    <arg name="sensor_parent_frame" default="photoneo_center_v1_optical_frame" />
    <arg name="topic_name" default="photoneo_center_v1" />
    <arg name="object_name" default="HV6" />
    <arg name="object_up" default="HV6_up" />
    <arg name="init_pose_up" default="-x -1 -y 0 -z 0.5 -R 0.0 -Y -0.0 -Z 0.0" />
    <arg name="init_x" default="-1" />
    <include file="$(find gen_dataset)/launch/gen_object.launch" >
      <arg name="object_name" value="$(arg object_name)" />
      <arg name="object_up" value="$(arg object_up)" />
      <arg name="init_pose_up" value="$(arg init_pose_up)" />
    </include>
    <node pkg="gen_dataset" type="random_move_euler.py" name="random_move_euler" output="screen" >
      <param name="object_name" value="$(arg object_name)" />
      <param name="num_dataset" value="$(arg num_dataset)" />
      <param name="topic_name" value="$(arg topic_name)" />
      <param name="object_up" value="$(arg object_up)" />
      <param name="init_x" value="$(arg init_x)" />
    </node>
    <node pkg="gen_dataset" type="record_data.py" name="record_data" output="screen" >
      <param name="sensor_parent_frame" value="$(arg sensor_parent_frame)" />
      <param name="topic_name" value="$(arg topic_name)" />
      <param name="object_name" value="$(arg object_name)" />
      <param name="num_dataset" value="$(arg num_dataset)" />
    </node>
    <node pkg="gen_dataset" type="odom_tf_converter" name="odom_tf_converter" output="screen" >
      <param name="object_name" value="$(arg object_name)" />
    </node>
  </group>

  <group ns="p2">
    <arg name="sensor_parent_frame" default="photoneo_center_v2_optical_frame" />
    <arg name="topic_name" default="photoneo_center_v2" />
    <arg name="object_name" default="HV7" />
    <arg name="object_up" default="HV7_up" />
    <arg name="init_pose_up" default="-x 0 -y 0 -z 0.5 -R 0.0 -Y -0.0 -Z 0.0" />
    <arg name="init_x" default="0" />
    <include file="$(find gen_dataset)/launch/gen_object.launch" >
      <arg name="object_name" value="$(arg object_name)" />
      <arg name="object_up" value="$(arg object_up)" />
      <arg name="init_pose_up" value="$(arg init_pose_up)" />
    </include>
    <node pkg="gen_dataset" type="random_move_euler.py" name="random_move_euler" output="screen" >
      <param name="object_name" value="$(arg object_name)" />
      <param name="num_dataset" value="$(arg num_dataset)" />
      <param name="sensor_parent_frame" value="$(arg sensor_parent_frame)" />
      <param name="topic_name" value="$(arg topic_name)" />
      <param name="object_up" value="$(arg object_up)" />
      <param name="init_x" value="$(arg init_x)" />
    </node>
    <node pkg="gen_dataset" type="record_data.py" name="record_data" output="screen" >
      <param name="sensor_parent_frame" value="$(arg sensor_parent_frame)" />
      <param name="topic_name" value="$(arg topic_name)" />
      <param name="object_name" value="$(arg object_name)" />
      <param name="num_dataset" value="$(arg num_dataset)" />
    </node>
    <node pkg="gen_dataset" type="odom_tf_converter" name="odom_tf_converter" output="screen" >
      <param name="object_name" value="$(arg object_name)" />
    </node>
    <node name="setup_commands" pkg="timed_roslaunch" type="timed_roslaunch.sh" args="3 gen_dataset setup_commands.launch" output="screen" />
  </group>

  <group ns="p3">
    <arg name="sensor_parent_frame" default="photoneo_center_v3_optical_frame" />
    <arg name="topic_name" default="photoneo_center_v3" />
    <arg name="object_name" default="HV8" />
    <arg name="object_up" default="HV8_up" />
    <arg name="init_pose_up" default="-x 1 -y 0 -z 0.5 -R 0.0 -Y -0.0 -Z 0.0" />
    <arg name="init_x" default="1" />
    <include file="$(find gen_dataset)/launch/gen_object.launch" >
      <arg name="object_name" value="$(arg object_name)" />
      <arg name="object_up" value="$(arg object_up)" />
      <arg name="init_pose_up" value="$(arg init_pose_up)" />
    </include>
    <node pkg="gen_dataset" type="random_move_euler.py" name="random_move_euler" output="screen" >
      <param name="object_name" value="$(arg object_name)" />
      <param name="num_dataset" value="$(arg num_dataset)" />
      <param name="sensor_parent_frame" value="$(arg sensor_parent_frame)" />
      <param name="topic_name" value="$(arg topic_name)" />
      <param name="object_up" value="$(arg object_up)" />
      <param name="init_x" value="$(arg init_x)" />
    </node>
    <node pkg="gen_dataset" type="record_data.py" name="record_data" output="screen" >
      <param name="sensor_parent_frame" value="$(arg sensor_parent_frame)" />
      <param name="topic_name" value="$(arg topic_name)" />
      <param name="object_name" value="$(arg object_name)" />
      <param name="num_dataset" value="$(arg num_dataset)" />
    </node>
    <node pkg="gen_dataset" type="odom_tf_converter" name="odom_tf_converter" output="screen" >
      <param name="object_name" value="$(arg object_name)" />
    </node>
    <node name="setup_commands" pkg="timed_roslaunch" type="timed_roslaunch.sh" args="3 gen_dataset setup_commands.launch" output="screen" />
  </group>
</launch>
